pkg/controller/petset/identity_mappers.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package petset
+
+import (
+	"crypto/md5"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	podapi "k8s.io/kubernetes/pkg/api/pod"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+// svcLocalSuffix is the search path for cluster local services.
+const svcLocalSuffix = "svc.cluster.local"
```

Comment:
It's just a const used in the string method because i wanted to print (and compare) the fqdn, the actual search path is chosen by kube-dns. I'll remove it since it can be confusing. Controller manager doesn't have a flag and I don't want to add one for this.